UC4-chaincode/src/main/java/de/upb/cs/uc4/chaincode/contract/ContractUtil.java

```diff
@@ -123,24 +122,25 @@ public void validateApprovals(
         }
 
         OperationContractUtil oUtil = new OperationContractUtil();
-        ApprovalList approvals;
         String key;
         try {
             key = OperationContractUtil.getDraftKey(contractName, transactionName, jsonArgs);
         } catch (NoSuchAlgorithmException e) {
             throw new ValidationError(GsonWrapper.toJson(getInternalError()));
         }
+        OperationData operation;
         try{
-            approvals = oUtil.getState(stub, key, OperationData.class).getExistingApprovals();
+            operation = oUtil.getState(stub, key, OperationData.class);
         } catch (Exception e) {
-            approvals = new ApprovalList();
+            throw new ValidationError(GsonWrapper.toJson(getInsufficientApprovalsError()));
         }
         String clientId = getEnrollmentIdFromClientId(ctx.getClientIdentity().getId());
         List<String> clientGroups = new GroupContractUtil().getGroupNamesForUser(ctx.getStub(), clientId);
+        ApprovalList approvals = operation.getExistingApprovals();
         approvals.addUsersItem(clientId);
         approvals.addGroupsItems(clientGroups);
```

Comment:
If no operation exists for a regular transaction, wouldn`t the code fail here?
I thought the default assumptions was: if no operation exists, we start at a new approvalList

Reply:
i suggest something like:
```java
ApprovalList approvals;
State state;
try{
    OperationData oData = oUtil.getState(stub, key, OperationData.class);
    approvals = oData.getExistingApprovals();
    operation = oData.getState();
} catch (Exception e) {
    approvals = new ApprovalList();
    state = OperationState.Pending
}
```

Reply:
Good point! Updated it, but there is currently no transaction to test this with.